Plugins/Format Source Code/plugin.swift

```diff
@@ -0,0 +1,93 @@
+import PackagePlugin
+import Foundation
+
+@main
+struct FormatSourceCode: CommandPlugin {
+  func performCommand(
+    context: PluginContext,
+    arguments: [String]
+  ) async throws {
+    let swiftFormatTool = try context.tool(named: "swift-format")
+    
+    // Extract the arguments that specify what targets to format.
+    var argExtractor = ArgumentExtractor(arguments)
+    let targetNames = argExtractor.extractOption(named: "target")
+    let targetsToFormat = try context.package.targets(named: targetNames)
+    
+    let configuration = argExtractor.extractOption(named: "configuration").first
+    
+    for target in targetsToFormat {
+      // Skip any type of target that doesn't have source files.
+      guard let target = target as? SourceModuleTarget else {
+        Diagnostics.warning("Skipping target “\(target.name)” because it doesn't contain Swift sources")
+        continue
+      }
+      
+      let swiftFormatExec = URL(fileURLWithPath: swiftFormatTool.path.string)
+      var swiftFormatArgs = [
+        "format",
+        target.directory.string,
+        "-r",
+        "-p",
+        "-i",
+      ]
+      
+      if let configuration {
```

Comment:
Here and elsewhere, let's avoid using new Swift 5.7 syntax for now; Package.swift still specifies a 5.6 tools version, and I'm not sure how it plays with CI (some other projects avoid it).

Reply:
I fixed it. https://github.com/zunda-pixel/swift-format/commit/60c66fc7a8404c4bbfd30506d38450042b292e5f